pkg/registry/core/rest/storage_core.go

```diff
@@ -269,9 +271,16 @@ func (s componentStatusStorage) serversToValidate() map[string]componentstatus.S
 			port = 2379
 		}
 		// TODO: etcd health checking should be abstracted in the storage tier
-		serversToValidate[fmt.Sprintf("etcd-%d", ix)] = componentstatus.Server{
+		var cert *tls.Certificate
+		if len(machine.CertFile) > 0 && len(machine.KeyFile) > 0 {
+			if cert, err = tlsutil.NewCert(machine.CertFile, machine.KeyFile, nil); err != nil {
```

Comment:
yeah, exactly, the original code comes from etcd-client code.. I made this change already.